pkg/cmd/bazci/bazci.go

```diff
@@ -284,10 +280,10 @@ func init() {
 }
 
 func bazciImpl(cmd *cobra.Command, args []string) (retErr error) {
-	var goTestJSONOutputFilePath string
+	var testXmls []string // We'll update this when the build is done.
 	defer func() {
-		if err := processTestJSONIfNeeded(retErr != nil /* shouldCreateTarball */, goTestJSONOutputFilePath); err != nil {
-			fmt.Printf("failed to process go test json output - %v\n", err)
+		if err := processTestXmls(testXmls); err != nil {
+			fmt.Printf("failed to process test.xml files - %+v\n", err)
 		}
```

Comment:
```suggestion
		if err := processTestXmls(testXmls); err != nil {
		  if retErr == nil {
		      retErr = errors.New(fmt.Sprintf("failed to process test.xml files - %+v\n", err))
		  }
		}
```

Reply:
Continuing from the suggestion below: shouldn't we fail the run if we fail to process test results so it doesn't go unnoticed?